app/Services/EDocument/Imports/ZugferdEDocument.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2024. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+namespace App\Services\EDocument\Imports;
+
+use App\Factory\ExpenseFactory;
+use App\Models\Currency;
+use App\Repositories\VendorRepository;
+use App\Services\AbstractService;
+use Exception;
+use horstoeko\zugferd\ZugferdDocumentReader;
+
+class ZugferdEDocument extends AbstractService
+{
+    public ZugferdDocumentReader $document;
+
+    /**
+     * @throws Exception
+     */
+    public function __construct(public string $tempdocument)
+    {
+        $this->document = ZugferdDocumentReader::readAndGuessFromContent($this->tempdocument);
+        # curl -X POST http://localhost:8000/api/v1/edocument/upload -H "Content-Type: multipart/form-data" -H "X-API-TOKEN: 7tdDdkz987H3AYIWhNGXy8jTjJIoDhkAclCDLE26cTCj1KYX7EBHC66VEitJwWhn" -H "X-Requested-With: XMLHttpRequest" -F _method=PUT -F documents[]=@einvoice.xml
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function run(): string
```

Comment:
please let it return the expense, so it can be reused by the InboundMailEngine :)

Reply:
Done